### PR TITLE
boot: group SealKeyModelParams by model, improve testing

### PR DESCRIPTION
**Pull request by cmatsuoka**
Create one set of model parameters per model, which will carry the
load chains and kernel command lines from all the boot chains that
reference that model.

This also improves testing.

### CODE REVIEW

**Comment by pedronis** [comment]
this uses #9325 now

File: boot/seal_test.go [hunk 1]
```diff
@@ -251,3 +265,105 @@ func createMockGrubCfg(baseDir string) error {
...
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(p, nil, 0644)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/loader-loader-hash1"), nil, 0644)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/loader-loader-hash2"), nil, 0644)
+	c.Assert(err, IsNil)
+
+	headers := model.Headers()
+	headers["model"] = "old-model-uc20"
```
Comment (mvo5) [inline boot/seal_test.go:292]: Thanks for adding this test

**Review by mvo5 — APPROVED** [review]
LGTM

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**